pyomo/contrib/doe/doe.py

```diff
@@ -767,12 +501,65 @@ def _direct_kaug(self):
                                     scale_constant_value=self.scale_constant_value)
         
         self.jac = jac
-        FIM_analysis.build_time = time_build
-        FIM_analysis.solve_time = time_solve
+        self.mod = mod
         
-
         return FIM_analysis
 
+    def _create_block(self):
+        """
+        Create a pyomo Concrete model and add blocks with different parameter perturbation scenarios. 
+        """
+
+        # create scenario information for block scenarios
+        scena_object = ScenarioGenerator(self.param, formula=self.formula, step=self.step)
```

Comment:
Proposal: Make the `__init__` method in `ScenarioGenerator` call `.generate_scenario()`. Then instead of returning the dictionary of dictionaries which is `scen_gen`, instead store these in the `scena_object` as `.parameter_values`, `.scenario_number`, and `scenario_perturbation`.

The nested dictionaries seem unnecessarily confusing.

Reply:
Also, rename `scena_object` as `scenario_data`. I know that `scena` is fewer characters, but the abbreviation makes the code harder to follow.